### PR TITLE
Added toc and refactored Combinational Logic

### PR DESCRIPTION
**Pull request by Dishebh**
#### Changes done:
- [x] Added toc and refactored Combinational Logic

Preview Link: https://deploy-preview-318--tender-ardinghelli-ebb79f.netlify.app/docs/Combinational/half_adder.html
Preview Link: https://deploy-preview-318--tender-ardinghelli-ebb79f.netlify.app/docs/Combinational/full_adder.html
Preview Link: https://deploy-preview-318--tender-ardinghelli-ebb79f.netlify.app/docs/Combinational/half_sub.html
Preview Link: https://deploy-preview-318--tender-ardinghelli-ebb79f.netlify.app/docs/Combinational/full_sub.html
Preview Link: https://deploy-preview-318--tender-ardinghelli-ebb79f.netlify.app/docs/Combinational/N-Bit%20Parallel%20Adder%20&%20Subtractor.html

#### ✅️ By submitting this PR, I have verified the following

- [x] Checked to see if a similar PR has already been opened 🤔️
- [x] Reviewed the contributing guidelines 🔍️
- [x] Sample preview link added (add a link from the checks tab after checks complete)
- [x] Tried Squashing the commits into one

### CODE REVIEW

**Comment by Dishebh** [comment]
> why does the sidebar looks different in all these pages?
> image

In what way?

**Comment by abhinavsri360** [comment]
I mean in other pages there are some lines in that button.

File: docs/Combinational/full_adder.md [hunk 1]
```diff
@@ -9,25 +9,30 @@ comments: true
...
+1. TOC
+{:toc}
+
+---
 
+## Introduction
 
+Full adder is developed to overcome the drawback of Half Adder circuit. 
+It can add two one-bit numbers A and B, and carry c. 
+The full adder is a three-input and two output combinational circuit.
```
Comment (YashKumarVerma) [inline docs/Combinational/full_adder.md:25]: directly picked from https://www.tutorialspoint.com/computer_logical_organization/combinational_circuits.htm

**Comment by Shivansh2407** [comment]
> I mean in other pages there are some lines in that button. Whereas in this one the button looks to be a little different.

This is an issue which is fixed in #237 . It will work after merging that PR.

**Comment by Dishebh** [comment]
> Looks good to me, but one thing. What's the advantage of a refactor when the contents are still a copy from other sources. Therefore I'd recommend doing one thing in one PR.
> 
> If you are adding a table of contents (~toc~, it's ambiguous), add it in one PR.
> plus, how exactly do you **refactor** a text article? consider using phrases like fixing formatting irregularities etc.

Sure, will keep that in mind from now. Thanks!

**Pull request merged**